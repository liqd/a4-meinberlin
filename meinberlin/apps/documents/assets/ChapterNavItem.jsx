var React = require('react')
var django = require('django')
var classNames = require('classnames')

function getErrorCount (props) {
  if (props.errors && Object.keys(props.errors).length > 0) {
    let errorCount = Object.keys(props.errors).length
    if (props.errors['paragraphs']) {
      errorCount = errorCount - 1 + Object.keys(props.errors['paragraphs']).length
    }
    return <span className="u-danger"> ({errorCount})</span>
  }
}

const ChapterNavItem = (props) => {
  return (
    <div className="commenting">
      <button
        type="button"
        className={classNames('commenting__content', 'commenting--toc__button', 'btn btn--light', 'btn--small', { 'active': props.active })}
        onClick={props.onClick}>
        {props.index + 1}. {props.name}
        {getErrorCount(props)}
      </button>

      <div className="commenting__actions btn-group" role="group">
        <button
          className="btn btn--light btn--small"
          onClick={props.onMoveUp}
          disabled={!props.onMoveUp}
          title={django.gettext('Move up')}
          type="button">
          <i className="fa fa-chevron-up"
            aria-label={django.gettext('Move up')} />
        </button>
        <button
          className="btn btn--light btn--small"
          onClick={props.onMoveDown}
          disabled={!props.onMoveDown}
          title={django.gettext('Move down')}
          type="button">
          <i className="fa fa-chevron-down"
            aria-label={django.gettext('Move down')} />
        </button>
        <button
          className="btn btn--light btn--small"
          onClick={props.onDelete}
          disabled={!props.onDelete}
          title={django.gettext('Delete')}
          type="button">
          <i className="fas fa-trash-alt"
            aria-label={django.gettext('Delete')} />
        </button>
      </div>
    </div>
  )
}

module.exports = ChapterNavItem
