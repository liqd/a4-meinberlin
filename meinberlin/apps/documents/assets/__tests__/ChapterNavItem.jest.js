// tools needed for testing
import React from 'react'
import { render } from '@testing-library/react'
import ChapterNavItem from '../ChapterNavItem'

test('ChapterNavItem top renders correctly', () => {
  const { asFragment } = render(
    <ChapterNavItem
      name="Chapter name"
      index={0}
      active
      onMoveUp={null}
    />)
  expect(asFragment()).toMatchSnapshot()
})

test('ChapterNavItem middle renders correctly', () => {
  const { asFragment } = render(
    <ChapterNavItem
      name="Chapter name"
      index={2}
      active={false}
    />)
  expect(asFragment()).toMatchSnapshot()
})

test('ChapterNavItem bottom renders correctly', () => {
  const { asFragment } = render(
    <ChapterNavItem
      name="Chapter name"
      index={3}
      active={false}
      onMoveDown={null}
    />)
  expect(asFragment()).toMatchSnapshot()
})
