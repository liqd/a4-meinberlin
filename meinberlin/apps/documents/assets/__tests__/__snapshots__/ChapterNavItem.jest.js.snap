// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChapterNavItem bottom renders correctly 1`] = `
<DocumentFragment>
  <div
    class="commenting"
  >
    <button
      class="commenting__content commenting--toc__button btn btn--light btn--small"
      type="button"
    >
      4. Chapter name
    </button>
    <div
      class="commenting__actions btn-group"
      role="group"
    >
      <button
        class="btn btn--light btn--small"
        disabled=""
        title="mock text"
        type="button"
      >
        <i
          aria-label="mock text"
          class="fa fa-chevron-up"
        />
      </button>
      <button
        class="btn btn--light btn--small"
        disabled=""
        title="mock text"
        type="button"
      >
        <i
          aria-label="mock text"
          class="fa fa-chevron-down"
        />
      </button>
      <button
        class="btn btn--light btn--small"
        disabled=""
        title="mock text"
        type="button"
      >
        <i
          aria-label="mock text"
          class="fas fa-trash-alt"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ChapterNavItem middle renders correctly 1`] = `
<DocumentFragment>
  <div
    class="commenting"
  >
    <button
      class="commenting__content commenting--toc__button btn btn--light btn--small"
      type="button"
    >
      3. Chapter name
    </button>
    <div
      class="commenting__actions btn-group"
      role="group"
    >
      <button
        class="btn btn--light btn--small"
        disabled=""
        title="mock text"
        type="button"
      >
        <i
          aria-label="mock text"
          class="fa fa-chevron-up"
        />
      </button>
      <button
        class="btn btn--light btn--small"
        disabled=""
        title="mock text"
        type="button"
      >
        <i
          aria-label="mock text"
          class="fa fa-chevron-down"
        />
      </button>
      <button
        class="btn btn--light btn--small"
        disabled=""
        title="mock text"
        type="button"
      >
        <i
          aria-label="mock text"
          class="fas fa-trash-alt"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ChapterNavItem top renders correctly 1`] = `
<DocumentFragment>
  <div
    class="commenting"
  >
    <button
      class="commenting__content commenting--toc__button btn btn--light btn--small active"
      type="button"
    >
      1. Chapter name
    </button>
    <div
      class="commenting__actions btn-group"
      role="group"
    >
      <button
        class="btn btn--light btn--small"
        disabled=""
        title="mock text"
        type="button"
      >
        <i
          aria-label="mock text"
          class="fa fa-chevron-up"
        />
      </button>
      <button
        class="btn btn--light btn--small"
        disabled=""
        title="mock text"
        type="button"
      >
        <i
          aria-label="mock text"
          class="fa fa-chevron-down"
        />
      </button>
      <button
        class="btn btn--light btn--small"
        disabled=""
        title="mock text"
        type="button"
      >
        <i
          aria-label="mock text"
          class="fas fa-trash-alt"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
